src/devices/Arduino/README.md

```diff
@@ -0,0 +1,90 @@
+- Go to C:\users\<username>\documents\arduino\libraries and delete the "ConfigurableFirmata" folder (save any work if you've changed anything there)
+- Replace it with a clone of https://github.com/pgrawehr/ConfigurableFirmata and switch to branch "develop". 
+- Make sure you have the "DHT Sensor Library" from Adafruit installed (use the library manager for that).
+- You can now enable the DHT and SPI features at the beginning of the ConfigurableFirmata.ino file.
+- Compile and re-upload the sketch. 
+
+## Usage
+See the example for advanced usage instructions. 
+
+Basic start:
+```
+            // Portname is "COM3", "COM4" on Windows, "/dev/ttyUSB0" or similar on linux
+            using (var port = new SerialPort(portName, 115200))
+            {
+                Console.WriteLine($"Connecting to Arduino on {portName}");
+                try
+                {
+                    port.Open();
+                }
+                catch (UnauthorizedAccessException x)
+                {
+                    Console.WriteLine($"Could not open COM port: {x.Message} Possible reason: Arduino IDE connected or serial console open");
+                    return;
+                }
+
+                ArduinoBoard board = new ArduinoBoard(port.BaseStream);
+                try
+                {
+                    board.LogMessages += BoardOnLogMessages; // Get log messages
+                    board.Initialize();
+                    Console.WriteLine($"Connection successful. Firmware version: {board.FirmwareVersion}, Builder: {board.FirmwareName}");
+                    // Add code that uses the board here.
+                }
+                catch (TimeoutException x)
+                {
+                    Console.WriteLine($"No answer from board: {x.Message} ");
+                }
+                finally
+                {
+                    port.Close();
+                    board?.Dispose();
+                }
+            }
+ ```
+ 
+On Windows, only one application can use the serial port at a time, therefore you'll get "permission denied" errors when you try to run your program while i.e. the Serial Port Monitor of the Arduino IDE is open when you start your program. On the other hand, trying to upload a new sketch while the program runs will also fail.
+
+## Known limitations
+
+All communication is routed trough the USB cable immitating a serial port with a limited bandwith. Therefore, some not insignificant delays are to be expected when sending commands or retrieving data. Communicating with sensors which have time-critical behavior will most likely not work reliably for this reason and the standard bindings provided for these won't work. This includes sensors like the DHT11, DHT22 or HCSR-04. For some of these, special Firmata modules are available to execute the time-critical part directly on the Arduino. This problem does not exist for sensors using I2C or SPI protocols. 
```

Comment:
If this is all USB interaction then `usb` category should not be here. It's more serial port related than USB. USB is incidental here.

Reply:
See above